### PR TITLE
fix(spacing): remove mb prop and document atomic spacing classes

### PR DESCRIPTION
**Pull request by moshie**
Removed the mb prop from `<Heading />`, `<Text />` & `<Badge />`

Update migration method and component documentation.

### CODE REVIEW

File: etc/docs/migration-v4.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+# Overview 🔭
+
+## What has changed?
+
+### 📐 `Spacing`
+
+With V4 introduces a new way of spacing components apart. Using atomic classes there is more granular control over spacing at different breakpoints.
+
+As part of this upgrade the `mb` property on `<Heading />`, `<Text />` & `<Badge />` components has been deprecated.
```
Comment (lluia) [inline etc/docs/migration-v4.md:20]: `has been removed.` as they won't be able to use it anymore in **v4** 👍🏻

File: etc/docs/migration-v4.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+Now you can use className to define different margin values
+
+```tsx static
+<Text as="p" className="mb-6">
+  This is a text element
+</Text>
+```
+
+Note the `as="p"` spans are natuarly inline elements and margin will **not** affect them
```
Comment (lluia) [inline etc/docs/migration-v4.md:36]: `as="span"` 💅🏻

File: src/README.md [hunk 3]
```diff
@@ -60,6 +60,8 @@ Margin: https://tailwindcss.com/docs/margin/
 
 Padding: https://tailwindcss.com/docs/padding/
 
+The introduction of this new feature means the **mb** property on `<Heading />` & `<Text />` are now deprecated.
```
Comment (lluia) [inline src/README.md:63]: I think we won't need this given we already have the **migration guide** documenting this change ✍🏻 👍🏻

Same for the docs of `<Heading />` and `<Text />`.

File: src/components/atoms/Text/Text.tsx [hunk 4]
```diff
@@ -53,13 +48,6 @@ const Text = styled.span<ITextProps>`
   letter-spacing: 0;
   color: ${({ color = colors.greyDarkest }) => color};
 
-  ${({ mb = false }) =>
-    mb &&
-    `
-    display: block;
-    margin-bottom:   24px
-  `};
```
Comment (lluia) [inline src/components/atoms/Text/Text.tsx:61]: I wonder if we could add `display: block` by parsing `className`:
```js
${({ className, as }) => as === 'span' && className.split(' ').some(clss => clss.includes('mb')) && `
  display: block
`}
``` 
The reasoning being `<Text />` by default renders as a `<span />` and it won't be nice for clients to have to do wrapper components every time:
```js
const BlockText = styled(Text)`display: block`
<BlockText className="mb-6" />
```

File: src/components/atoms/Text/Text.tsx [hunk 5]
```diff
@@ -53,20 +48,20 @@ const Text = styled.span<ITextProps>`
...
   line-height: ${({ size = 'body' }) => lineHeightMap[size]};
   font-weight: ${({ weight = 'regular', capitalize }) => typography.weights[capitalize ? 'bold' : weight]};
 
   font-family: ${typography.primary};
   text-align: ${({ align = 'inherit' }) => align};
   text-transform: ${({ capitalize }) => capitalize && 'uppercase'};
+
+  ${({ className = '', as }) =>
+    (!as || as === 'span') &&
+    className.split(' ').some(clss => clss.includes('mb')) &&
```
Comment (lluia) [inline src/components/atoms/Text/Text.tsx:61]: We probably are gonna need to handle more cases than just `mb`, lets catch-up later 👍🏻

Comment (moshie) [reply src/components/atoms/Text/Text.tsx:61]: @lluia 

How do you feel about adding some regex to detect if a spacing class is being used for example: 

```js
/[mp](\:[mp])?[tblr]?-\d+/g
```

**Pull request merged**